backend/app/routers/feeds.py

```diff
@@ -0,0 +1,158 @@
+from typing import List
+import os
+from bson import ObjectId
+from fastapi import APIRouter, HTTPException, Depends, Request
+from pymongo import MongoClient
+import datetime
+from app.dependencies import get_db
+from app.keycloak_auth import get_user, get_current_user
+from app.models.users import UserOut
+from app.models.files import FileOut
+from app.models.listeners import (
+    FeedListener,
+    EventListenerOut,
+)
+from app.models.feeds import (
+    FeedIn,
+    FeedDB,
+    FeedOut,
+)
+from app.models.search import SearchIndexContents
+from app.search.connect import check_search_result
+from app.rabbitmq.listeners import submit_file_message
+
+router = APIRouter()
+
+clowder_bucket = os.getenv("MINIO_BUCKET_NAME", "clowder")
+
+
+# TODO: Move this to MongoDB middle layer
+async def disassociate_listener_db(feed_id: str, listener_id: str, db: MongoClient):
+    """Remove a specific Event Listener from a feed. Does not delete either resource, just removes relationship.
+
+    This actually performs the database operations, and can be used by any endpoints that need this functionality."""
+    async for feed in db["feeds"].find(
+        {"listeners.listener_id": ObjectId(listener_id)}
+    ):
+        feed_db = FeedDB.from_mongo(feed)
+        new_listeners = []
+        for feed_listener in feed_db.listeners:
+            if feed_listener.listener_id != listener_id:
+                new_listeners.append(feed_listener)
+        feed_db.listeners = new_listeners
+        await db["feeds"].replace_one(
+            {"_id": ObjectId(feed_id)}, FeedDB(**feed_db).to_mongo()
+        )
+
+
+async def check_feed_listeners(
+    es_client,
+    file_out: FileOut,
+    user: UserOut,
+    db: MongoClient,
+):
+    """Automatically submit new file to listeners on feeds that fit the search criteria."""
+    listeners_found = []
+    async for feed in db["feeds"].find({"listeners": {"$ne": []}}):
+        feed_db = FeedDB(**feed)
+
+        # If feed doesn't have any auto-triggering listeners, we're done
+        found_auto = False
+        for listener in feed_db.listeners:
+            if listener.automatic:
+                found_auto = True
+                break
+
+        if found_auto:
+            # Verify whether resource_id is found when searching the specified criteria
+            feed_match = check_search_result(es_client, file_out, feed_db.search)
+            if feed_match:
+                for listener in feed_db.listeners:
+                    if listener.automatic:
+                        listeners_found.append(listener.listener_id)
+
+    for targ_listener in listeners_found:
+        queue = ""  # TODO: Each extractor gets a queue - routing key same as name?
```

Comment:
Can this default to the listener name so that legacy extractors can be triggered?